salesagility/SuiteCRM
# Fix #9649 - Inheritance of the parent record does not work when creating a record by code

closes #9649 

## Description





The application of the inheritance rules of the parent record are in the function `inherit_parent()`, in _modules/securitygroups/securitygroup.php_ file. This function loops the relationships of the new registration to recover the different security groups from related records, using $_REQUEST to determine the value of related fields, but $_REQUEST does not include this information in various situations such as record creation via code.

## Motivation and Context

This PR rewrites several parts of the `inherit_parent()` function to include the inheritance of the records saving via code.

## How To Test This

1. Enable "Inherit from parent record" in Security Suite settings and disable "Inherit from Created By User" and "Inherit from Assigned To User"
2. Create an Security Group. (Ex. SG1)
3. Create an Account record and assign SG1
4. Create a Contact record (or another module) from the subpanel, in the Account detail view.
5. Check that the contacts record has inherited the SG1 security group
6. Create a Contacts record through programming, you can use the following code on a after_save logic hook of accounts:
    ```php
    <?php

    class AccountsLogicHooks
    {
        public function after_save(&$bean, $event, $arguments)
        {
            $bean->load_relationships('contacts');
            $contactBean = BeanFactory::newBean('Contacts');
            $contactBean->last_name = 'New contact created';
            $contactBean->save();
            $bean->contacts->add($contactBean);
        }
    }
    ```

7. After executing the action launched by logic_hook, check that the contacts record has inherited the SG1 security group.
8. Reproduce in some more cases the creation of records via code and verify that the inheritance of the security groups of the parent Record behaves just like if it was done from the SuiteCRM interface.



## Types of changes


### Final checklist
